src/MallocBuffer.hpp

```diff
@@ -130,16 +130,14 @@ class MallocBuffer : public bufferManipulation::VoidBuffer
   void reallocate( std::ptrdiff_t const size, MemorySpace const space, std::ptrdiff_t const newCapacity )
   {
     LVARRAY_ERROR_IF_NE( space, MemorySpace::host );
-
-#if (defined(__GNUC__) && (__GNUC__ == 8))
-    // This is a workaround for a GCC bug https://gcc.gnu.org/bugzilla/show_bug.cgi?id=87544
-    std::size_t constexpr maxSize = (NumericLimits< std::size_t >::max / sizeof( T )) >> 1;
-    std::size_t newSpaceSize = math::min( static_cast< std::size_t >(newCapacity), maxSize ) * sizeof( T );
-#else
     std::size_t newSpaceSize = newCapacity * sizeof( T );
-#endif
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Walloc-size-larger-than="
```

Comment:
they claim that the bug was fixed in GCC 9. Are we sure that it's a compiler bug and not an actual issue?

Reply:
The problem keeps appearing, for example see [here](https://github.com/ocornut/imgui/issues/5343).

Reply:
A simple reproducer found [here](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105651):

```c++
#include <string>

void f(const long long v)
{
   std::string line;
   line += " " + std::to_string(v);
}
```

```
$ g++ -Wall -O3 -std=c++20 -o test_warning.o -c test_warning.cxx
```

On quartz with `GCC 12.1.1` gives:

```
[...]
    inlined from ‘void f(long long int)’ at test_warning.cpp:6:16:
/collab/usr/global/tools/tce4/packages/gcc/gcc-12.1.1/include/c++/12/bits/char_traits.h:431:56: warning: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ accessing 9223372036854775810 or more bytes at offsets [2, 9223372036854775807] and 1 may overlap up to 9223372036854775813 bytes at offset -3 [-Wrestrict]
  431 |         return static_cast<char_type*>(__builtin_memcpy(__s1, __s2, __n));
```

Reply:
@castelletto1 That is a different warning `-Wrestict`